Doc/library/typing.rst

```diff
@@ -1230,6 +1232,83 @@ These are not used in annotations. They are building blocks for creating generic
     ``covariant=True`` or ``contravariant=True``.  See :pep:`484` for more
     details.  By default, type variables are invariant.
 
+.. class:: TypeVarTuple
+
+    :class:`Type variable <TypeVar>` tuple.
+
+    Usage::
+
+        Ts = TypeVarTuple('Ts')
+        Shape = TypeVarTuple('Shape')
+
+    Vanilla type variables enable parameterization with a *fixed* number of
+    types. Type variable tuples, on the other hand, allow parameterization with
+    an *arbitrary* number of types by acting like an *arbitrary* number of type
+    variables wrapped in a tuple. For example::
+
+        class Array(Generic[*Shape]):
+            pass
+
+        from typing import NewType
+        Height = NewType('Height', int)
+        Width = NewType('Width', int)
+
+        1d_array: Array[Height] = Array()  # This is fine
+
+    With one type argument, ``Array`` behaves as if ``Shape = (T,)``, where
+    ``T`` is a type variable. That means ``Array`` behaves as if we'd written
+    ``Generic[T]``. Note the use of the unpacking operator ``*`` to bring ``T``
+    out of the tuple and directly into the square brackets.
+
+    ::
+
+        2d_array: Array[Height, Width] = Array()  # This is also fine!
+
+    With two arguments, it's as if ``Shape = (T1, T2)``, where ``T1`` and ``T2``
+    are type variables; as if ``Array`` was defined with ``Generic[T1, T2]``.
+
+    Type variable tuples can be used any place a vanilla type variable can.
```

Comment:
Not exactly, because I can write `def f(x: T) -> T:` but not `def f(x: *Ts) -> *Ts:`.

Reply:
Ah, you're right, we can't _quite_ say this. I've reworded this as "can be used in the same contexts", and added some simple variable annotation examples differentiating between `Ts`, `*Ts`, and `tuple[*Ts]`.